src/library/masq.js

```diff
@@ -271,6 +270,24 @@ class Masq {
     this._setProfileToLocalStorage(updatePublicProfile)
   }
 
+  /**
+   * Update passphrase
+   * @param {string} currentPassPhrase The current passphrase
+   * @param {string} newPassPhrase The new passphrase
+   */
+  async updatePassphrase (currentPassPhrase, newPassPhrase) {
+    this._checkProfile()
+    const protectedMK = await this._getProtectedMK()
+    try {
+      const protectedMKNewPass = await updateMasterKeyAndNonce(currentPassPhrase, newPassPhrase, protectedMK)
+      // we do not use this.encryptAndPut because this value is not encrypted
+      await this.profileDB.putAsync('/profile/protectedMK', protectedMKNewPass)
+    } catch (error) {
+      this.closeProfile()
```

Comment:
I think we should not close the profile, only throw. This way we will be able to display an error and the user will be able to retry to update his pass

Reply:
:+1: